atomix/cluster/src/test/java/io/atomix/raft/RaftStartupConsistencyCheckTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.atomix.raft;
+
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+import io.atomix.raft.snapshot.TestSnapshotStore;
+import java.util.concurrent.CompletableFuture;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class RaftStartupConsistencyCheckTest {
+
+  @Rule public RaftRule raftRule = RaftRule.withBootstrappedNodes(3);
+
+  @Test // Regression test for https://github.com/camunda/zeebe/issues/10451
+  public void shouldNotFailRestartIfFollowerCrashedBeforeCommittingSnapshot() throws Exception {
```

Comment:
I added this test for reproducing the bug. I'm not fully satisfied with this test. May be the unit test `StateUtilTest` is enough, a we can remove this one?

Reply:
I would keep the test :+1: I like it. I guess would be also great if we build us some more tooling to start nodes in a specific state, like here I want to start a follower with no log at certain index and no snapshot or something like that.